samples/features/sql-big-data-cluster/connectivity/webhdfs-java-client/src/main/java/com/microsoft/mssql/App.java

```diff
@@ -0,0 +1,118 @@
+package com.microsoft.mssql;
+
+import java.security.Principal;
+import java.io.File;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.Credentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
+/**
+ * Sample app for Encryption Zone interactions with WebHdfs.
+ */
+public class App {
+	
+	// Properties to be configured before running the application.
+	private static String GW_ENDPOINT = "https://knox.test.azdata.local:30443/gateway/default/webhdfs/v1";
+	
+	private static String KDC = "winad.azdata.local";
+	
+	private static String DOMAIN_REALM = "AZDATA.LOCAL";
+	
+	// The existing login.conf along with this project can be modified instead of providing a new path.
+	private static String LOGIN_CONF_PATH = "login.conf";
+	// End of properties to be configured.
+
+	static Credentials emptyCredentials = new Credentials() {
+		public String getPassword() {
+			return null;
+		}
+		public Principal getUserPrincipal() {
+			return null;
+		}
+	};
+	
+	static CredentialsProvider provider = new CredentialsProvider() {
+		@Override
+		public void setCredentials(AuthScope authscope, Credentials credentials) {
+			// No op
+		}
+
+		@Override
+		public Credentials getCredentials(AuthScope authscope) {
+			//TODO: this can be enhanced to pass controller use name and password in case the AuthScope authscheme is basic.
+			return emptyCredentials;
+		}
+
+		@Override
+		public void clear() {
+			// No op. Nothing to clean up.
+		}
+	};
+        private static void uploadResource(String inputFilePath, String outputFilePath){
+        	System.out.println("Entering create resource");
+                // Construct Knox endpoint
+                String createOperationEndpoint = GW_ENDPOINT + outputFilePath + "?op=CREATE&overwrite=true"; 
```

Comment:
GW_ENDPOINT is guaranteed to be different for the user. Perhaps take it as a command line input parameter?